react/OrderItems.tsx

```diff
@@ -427,39 +453,35 @@ export const OrderItemsProvider: FC = ({ children }) => {
       items: Array<Partial<CatalogItem>>,
       marketingData?: Partial<MarketingData>
     ) => {
-      const mutationInputItems = items
-        .map(adjustForItemInput)
-        .filter(
-          itemInput =>
-            orderFormItemsRef.current.findIndex(
-              orderFormItem => itemInput.id!.toString() === orderFormItem.id
-            ) === -1
-        )
+      const updatedItems = items.map((item) => {
+        const existingItem = findExistingItem(item, orderFormItemsRef.current)
 
-      const orderFormItems = mutationInputItems
-        .map(itemInput => {
-          const index = items.findIndex(
-            item => item.id === itemInput.id?.toString()
-          )
-          return mapItemInputToOrderFormItem(itemInput, items[index])
-        })
-        .filter(
-          orderFormItem =>
-            orderFormItemsRef.current.findIndex(
-              (item: any) => item.id === orderFormItem.id
-            ) === -1
-        )
+        if (existingItem == null) {
+          return item
+        }
+
+        return {
+          ...item,
+          id: item.id ?? '0',
+          quantity: (item.quantity ?? 1) + existingItem.quantity,
+        }
+      })
+
+      const mutationInputItems = updatedItems.map(adjustForItemInput)
+      const orderFormItems = updatedItems.map(mapToOrderFormItem)
 
       if (orderFormItems.length === 0) {
-        // all items already exist in the cart
         return false
```

Comment:
I guess we can remove the return value now, since we will always add the items to the cart, independently if they were already in it or not